src/plonkish/compiler/unit.rs

```diff
@@ -48,6 +48,8 @@ pub struct CompilationUnit<F> {
 
     pub other_sub_circuits: Rc<Vec<CompilationUnit<F>>>,
     pub other_columns: Rc<Vec<Column>>,
+
+    pub compilation_phase: usize,
```

Comment:
@rutefig we use usize for something that indexes an array or is the size of something, perhaps it should be u32. But I actually I would do phase1_complete:bool. Because we are not thinking in having more phases.

Reply:
I agree we can change to u32 for convention, but I think it's better to have the phase tracked anyways because it can give more flexibility in the future. Even if we don't have any more phases, we can track at some point that phase 2 is also already completed